### ISSUE TITLE
airspec: custom assertion

### ISSUE THREAD

**Issue by exoego** [wvlet/airframe#915]
It would be great if airspec support **_custom assertion_** feature as in other test frameworks.

One of my usecase of custom assertion is verifying equality of objects in JavaScript world on Scala.js.
Unfortunately, `==` operator does not return `true` on a pair of JS objects whose contents are identical, unlike Scala objects (e.g. `Seq(1) == Seq(1)`).
It is ofcourse possible to check content-equality when user define a predicate function something like this:

```scala
def contentEqual(a: js.Array[_], b: js.Array[_]): Boolean = {
  a.length == b.length && (a zip b).forall(_._1 == _._2)
}

test("js array") {
  val actual: js.Array[Int] = getA()
  assert(contentEqual(actual, Array(1, 2, 3)) 
}
```

But this is not helpful on test failure, because it is unclear which element is wrong.
As an user, I expect helpful failure message like `[4,5,6] is not equal to [1,2,3]`, which helps debugging.

This feature request may be related to
> Power assertion for test failures (e.g., show the condition expression and data differences)
#839

**Comment by xerial** [comment]
How about adding a support for js object comparison to shouldBe syntax? Then we can also enhance it to show the actual diff when implementing #839

**Comment by exoego** [comment]
That sounds good to me 👍 

Actually, I had wrote and published a custom assertion for ScalaTest.
[It allows comparing `js.Object` or `js.Array` based on their properties and contents](https://github.com/exoego/scalajs-test-helper/blob/92dd3ba77682294fd978335244e3f05f45c723af/module/scalatest/src/test/scala/net/exoego/scalajs/scalatest/JsObjectStructuralEqualityTest.scala).
[My implementation](https://github.com/exoego/scalajs-test-helper/blob/master/module/core/src/main/scala/net/exoego/scalajs/StructuralEqual.scala) might be helpful.
It works well for my own usage right now, but may miss some corner cases...

**Comment by xerial** [comment]
That's helpful. Thanks!

**Comment by xerial** [comment]
Ok. #922 is a PR for this and we need to wait Scala.js 1.0.0 release to use `js.Object.entries` method https://github.com/scala-js/scala-js/pull/3869

**Comment by xerial** [comment]
@exoego This feature is included in Airframe 20.2.0 along with Scala.js 1.0.0 support. Thanks!
